chainer/training/extensions/log_report.py

```diff
@@ -109,13 +121,36 @@ def __call__(self, trainer):
             # write to the log file
             if self._log_name is not None:
                 log_name = self._log_name.format(**stats_cpu)
-                with utils.tempdir(prefix=log_name, dir=trainer.out) as tempd:
-                    path = os.path.join(tempd, 'log.json')
-                    with open(path, 'w') as f:
-                        json.dump(self._log, f, indent=4)
 
-                    new_path = os.path.join(trainer.out, log_name)
-                    shutil.move(path, new_path)
+                if self._file_format == 'jsonl':
+                    # append the stats as a JSON object without CR/LF chars
+                    # and an LF as a delimiter
+                    stats_str = (json.dumps(stats_cpu, indent=None,
+                                            separators=(',', ':')) + '\n')
```

Comment:
Why using these separators, instead of the default?

Reply:
It is to remove whitespace as a suffix from the default separators (`, ` and `: `). It is the way suggested in the [official document](https://docs.python.org/3/library/json.html) (See _Compact encoding_).